src/wasm-lib/kcl/src/coredump/mod.rs

```diff
@@ -27,25 +32,24 @@ pub trait CoreDump: Clone {
 
     async fn get_webrtc_stats(&self) -> Result<WebrtcStats>;
 
+    async fn get_client_state(&self) -> Result<JValue>;
+
     /// Return a screenshot of the app.
     async fn screenshot(&self) -> Result<String>;
 
     /// Get a screenshot of the app and upload it to public cloud storage.
-    async fn upload_screenshot(&self) -> Result<String> {
+    async fn upload_screenshot(&self, coredump_id: &Uuid, zoo_client: &Client) -> Result<String> {
         let screenshot = self.screenshot().await?;
         let cleaned = screenshot.trim_start_matches("data:image/png;base64,");
-        // Create the zoo client.
-        let mut zoo = kittycad::Client::new(self.token()?);
-        zoo.set_base_url(&self.base_api_url()?);
 
         // Base64 decode the screenshot.
         let data = base64::engine::general_purpose::STANDARD.decode(cleaned)?;
         // Upload the screenshot.
-        let links = zoo
+        let links = zoo_client
             .meta()
             .create_debug_uploads(vec![kittycad::types::multipart::Attachment {
                 name: "".to_string(),
-                filename: Some("modeling-app/core-dump-screenshot.png".to_string()),
+                filename: Some(format!(r#"modeling-app/coredump-{}-screenshot.png"#, coredump_id)),
```

Comment:
```suggestion
                filename: Some(format!(r#"modeling-app/coredump-{coredump_id}-screenshot.png"#)),
```

nit: personally I think this is more readable